### PR TITLE
build: added a settlements schema

### LINKED ISSUE #808: Settlements Table Schema
## **Settlements**

Represents the overarching record of a financial activity. Each entry can have references to either a credit or a debit.

| Field    | Type         | Description                                    |
|----------|--------------|------------------------------------------------|
| `id`       | int8      | Unique identifier for the settlement.          |
| `created`  | timestamptz     | Timestamp when the settlement was created.     |
| `updated`  | timestamptz     | Timestamp when the settlement was last updated.|
| `user_id`       | int8      | GitHub `id` of the user penalized.                         | 139262667                      |
| `org_id`        | int8      | GitHub `id` of the organization.                           | 76412717                       |
| `repo_id`       | int8      | GitHub `id` of the repository where the issue is located.  | 534616569                      |
| `issue_id`      | int8      | GitHub `id` of the issue related to the penalty.           | 1868425596                     |
| `credit_id` | int8      | Foreign key to the Credits table.              |
| `debit_id`  | int8      | Foreign key to the Debits table.               |

### PR DESCRIPTION
**Pull request by GaryThisSidee**
Resolves #808

### CODE REVIEW

**Comment by seprintour** [comment]
Check comment on issue @GaryThisSidee

**Comment by 0x4007** [comment]
My mistake for not clarifying but the requirements inherit from the parent issue 

https://github.com/ubiquity/ubiquibot/issues/787

> The bot is expected to be stable after these changes are implemented which implies that all of the functions that interact with the database should also be refactored to accomodate this new architecture.

File: supabase/migrations/20230925111522_settlements.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+CREATE TABLE IF NOT EXISTS Settlements (
```
Comment (whilefoo) [inline supabase/migrations/20230925111522_settlements.sql:1]:
```suggestion
CREATE TABLE IF NOT EXISTS settlements (
```

File: supabase/migrations/20230925111522_settlements.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    id  serial PRIMARY KEY,
+    created  timestamptz NOT NULL,
+    updated  timestamptz NOT NULL,	
+    userId 	 int8 NOT NULL,
```
Comment (whilefoo) [inline supabase/migrations/20230925111522_settlements.sql:5]:
```suggestion
    user_id 	 int8 NOT NULL,
```

**Review by whilefoo — CHANGES_REQUESTED** [review]
please follow naming convention

**Comment by whilefoo** [comment]
@seprintour

**Comment by seprintour** [comment]
@whilefoo this is not my PR, it's incomplete anyway and it's supposed to be closed as the issue has been released to the devpool

**Comment by whilefoo** [comment]
oh sorry!

**Pull request closed without merging**